modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/sorting/Sorting.tsx

```diff
@@ -157,19 +157,68 @@ const AddFDSSortModalContent = ({
 			method: 'POST',
 		});
 
-		if (!response.ok) {
+		if (!addFDSSortResponse.ok) {
 			setSaveButtonDisabled(false);
 
 			openDefaultFailureToast();
 
 			return;
 		}
 
-		const responseJSON = await response.json();
+		const newFDSSort = await addFDSSortResponse.json();
+
+		// If an existing default sorting exists, change its default to false.
+
+		const defaultFDSSort = fdsSorts.find(
+			(fdsSort) => fdsSort.default
+		) as ISort;
```

Comment:
My biggest concern with this PR is the fact that we are taking previous default sort from `fdsSorts`, a local frontend state variable. What will happen if default state is changed in backend, while current user is browsing UI? Expected behavior would be that the latest default overrides the previous one. But with the solution in current PR, if I'm understanding it correctly, we will set `default: false` to an item that is no longer default. The patch request will be success, but we end up with two defaults in database, which may cause weird issues.

We could fetch latest all sorts and find default in there, but that only reduces the issue, does not fix it completely. It would still be possible that default is updated in the round-trip time between post response and patch request.

To ensure good state of defaults, we would need to update **all** sort entries, but I don't think that viable to do with rest API.

I see two solution paths that would address above:
1. We make an old-fashioned resource action, like we have for [saving table sections](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/java/com/liferay/frontend/data/set/admin/web/internal/portlet/action/SaveFDSFieldsMVCResourceCommand.java), and do all the updates in backend, in a single transaction. This would fix the double toast issue, as it would be a single request.
2. We store `defaultSort` in parent `FDSView` table (soon to be `DataSet`), instead of storing it in each sort item. We can still display it in each item, if we want to, through a custom cell renderer. Double toast issue would remain, but we would have a solution based only on headless API.

Reply:
That's a really good point I hadn't considered with the possibility of changes happening in the backend.

After considering your solutions, my current thoughts are that:
- Having a resource action is needed so that it can be a single transaction. Also fixes the double toast issue.
- I was also thinking of a combination of both solutions, but it _might_ be more straight-forward to keep the `default` value in `FDSSort` and in the resource action, if "Use as Default" is checked, it'll iterate through all the `FDSSort` objects to update the `default` values.

Out of curiosity, would you happen to know if these actions commands automatically handle rollbacks if there was a failure mid-transaction through the `BaseTransactionalMVCResourceCommand` or if there's additional code needed?

Reply:
Yep, `BaseTransactionalMVCResourceCommand` as a parent class ensures rollbacks, no additional code is needed.